src/pages/_app.tsx

```diff
@@ -57,14 +59,19 @@ const InitApp = (): null => {
 const clientSideEmotionCache = createEmotionCache()
 
 export const AppProviders = ({ children }: { children: ReactNode | ReactNode[] }) => {
-  const theme = useLightDarkTheme()
+  const isDarkMode = useDarkMode()
+  const themeMode = isDarkMode ? 'dark' : 'light'
 
   return (
-    <ThemeProvider theme={theme}>
-      <Sentry.ErrorBoundary showDialog fallback={ErrorBoundary}>
-        {children}
-      </Sentry.ErrorBoundary>
-    </ThemeProvider>
+    <SafeThemeProvider mode={themeMode}>
+      {(safeTheme: Theme) => (
```

Comment:
Question: why are we returning a provier and not a theme intitiation hook as we have in `web-core`? The dark mode hooks would likely benefit from existing in SRC as well, no?

Reply:
well, we are [exporting one indeed](https://github.com/safe-global/safe-react-components/blob/7b8929d497003d5905ec732485499b43ce8d80b1/src/hooks/useThemeMode.ts#L1-L36). We export the `createSafeTheme` as well but the `SafeThemeProvider` is convenient because creates it for you using the theme mode and return it in the render prop fn.

I'm using the hook you have instead of the exported one because there are custom work there setting the `data-theme`

Reply:
The provider is convenient due [this bug](https://github.com/mui/material-ui/issues/32806) (I added a new comment with our specific case). So it returns the children wrapping the `ThemeProvider` necessary for the components in the library to have access to the theme.